context.go

```diff
@@ -147,6 +150,20 @@ func WithProfileEvents(fn func([]ProfileEvent)) QueryOption {
 	}
 }
 
+func WithGotData(fn func()) QueryOption {
```

Comment:
I know the other options don't have this yet, but do you think we could add a `go doc` comment that explains that these functions are synchronous in the handler goroutine, so must not block?